webhook/resourcesemantics/defaulting/defaulting.go

```diff
@@ -70,6 +73,34 @@ type reconciler struct {
 	secretName            string
 }
 
+// CallbackFunc is the function to be invoked.
+type CallbackFunc func(ctx context.Context, unstructured *unstructured.Unstructured) error
+
+// Callback is a generic function to be called by a consumer of defaulting.
+type Callback struct {
+	// function is the callback to be invoked.
+	function CallbackFunc
+
+	// supportedVerbs are the verbs supported for the callback.
+	// The function will only be called on these actions.
+	supportedVerbs map[webhook.Operation]struct{}
+}
+
+// NewCallback creates a new callback function to be invoked on supported verbs.
+func NewCallback(function func(context.Context, *unstructured.Unstructured) error, supportedVerbs ...webhook.Operation) Callback {
```

Comment:
`Delete` operation wouldn't make sense here

Reply:
Yeah, the webhook will still be configured with:
```
			Operations: []admissionregistrationv1.OperationType{
				admissionregistrationv1.Create,
				admissionregistrationv1.Update,
			},
```

Are you suggesting to panic when a supportedVerbs is `Delete`?

Reply:
yeah - unless

1) There's a reason why someone would want DELETE
2) Does this actually work?

Reply:
Done